quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java

```diff
@@ -0,0 +1,52 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.SignupUserResponse;
+import com.upgrad.quora.api.model.SignupUserRequest;
+
+import com.upgrad.quora.service.Entity.UserEntity;
+import com.upgrad.quora.service.business.UserBusinessService;
+import com.upgrad.quora.service.exception.SignUpRestrictedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/")
+public class UserController {
+
+    @Autowired
+    private UserBusinessService userBusinessService;
+
+
+    //The userSignup()  method is of request post type , path is "/user/signup" the method set all the attribute of user in userEntity and generate a Response model with HTTP status indicating successfully registered
+    //The method also handle SignUpRestrictionException
+    @RequestMapping(method = RequestMethod.POST, path = "/user/signup", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<SignupUserResponse> userSignup(SignupUserRequest signupUserRequest) throws SignUpRestrictedException {
+
+        UserEntity userEntity = new UserEntity();
+
+        userEntity.setUuid(UUID.randomUUID().toString());
+        userEntity.setFirstName(signupUserRequest.getFirstName());
+        userEntity.setLastName(signupUserRequest.getLastName());
+        userEntity.setUsername(signupUserRequest.getUserName());
+        userEntity.setEmail(signupUserRequest.getEmailAddress());
+        userEntity.setPassword(signupUserRequest.getPassword());
+        userEntity.setContactNumber(signupUserRequest.getContactNumber());
+        userEntity.setAboutMe(signupUserRequest.getAboutMe());
+        userEntity.setDob(signupUserRequest.getDob());
+        userEntity.setCountry(signupUserRequest.getCountry());
+        userEntity.setSalt("1234abc");
```

Comment:
Should we hardcode salt and role? If we have to then declare it as private static variable

Reply:
1.  when password encryption is applied we are modifying  this variable as userEntity.setSalt(encryptedText[0]); so a method which is to be overridden or modify  cannot be declare static.
2. hard coding role is given in the instruction to set the role as "nonadmin" by default for all users